make/deploy.mk

```diff
@@ -0,0 +1,110 @@
+# On arm64, you can force the use of amd64 images:
+#
+#   make deploy CRI_ARCH=amd64
+#   make kind-load CRI_ARCH=amd64
+#
+CRI_ARCH ?= $(HOST_ARCH)
+
+# The commas in FEATURE_GATES have been kept for backwards compatibility with
+# our previous Bazel/scripts.
+FEATURE_GATES ?= AdditionalCertificateOutputFormats=true,ExperimentalCertificateSigningRequestControllers=true,ExperimentalGatewayAPISupport=true,ServerSideApply=true
+
+# In make, there is no way to escape commas or spaces. So we use the variables
+# $(space) and $(comma) instead.
+null  =
+space = $(null) #
+comma = ,
+
+FEATURE_GATES_CONTROLLER = $(subst $(space),$(comma),$(filter AllAlpha=% AllBeta=% AdditionalCertificateOutputFormats=% ValidateCAA=% ExperimentalCertificateSigningRequestControllers=% ExperimentalGatewayAPISupport=% ServerSideApply=%, $(subst $(comma),$(space),$(FEATURE_GATES))))
+FEATURE_GATES_WEBHOOK = $(subst $(space),$(comma),$(filter AllAlpha=% AllBeta=% AdditionalCertificateOutputFormats=% , $(subst $(comma),$(space),$(FEATURE_GATES))))
+FEATURE_GATES_CAINJECTOR = $(subst $(space),$(comma),$(filter AllAlpha=% AllBeta=%, $(subst $(comma),$(space),$(FEATURE_GATES))))
+
+# generated by ./hack/latest-kind-images.sh
+KIND_IMAGE_SHA_K8S_1.18=sha256:f4bcc97a0ad6e7abaf3f643d890add7efe6ee4ab90baeb374b4f41a4c95567eb
+KIND_IMAGE_SHA_K8S_1.19=sha256:a70639454e97a4b733f9d9b67e12c01f6b0297449d5b9cbbef87473458e26dca
+KIND_IMAGE_SHA_K8S_1.20=sha256:cbeaf907fc78ac97ce7b625e4bf0de16e3ea725daf6b04f930bd14c67c671ff9
+KIND_IMAGE_SHA_K8S_1.21=sha256:0fda882e43d425622f045b492f8bd83c2e0b4984fc03e2e05ec101ca1a685fb7
+KIND_IMAGE_SHA_K8S_1.22=sha256:f240c00ffb1d82a2a2225ca0f5c85d1c45aa2b97921327cb3f6da4eee7eae5c3
+KIND_IMAGE_SHA_K8S_1.23=sha256:49824ab1727c04e56a21a5d8372a402fcd32ea51ac96a2706a12af38934f81ac
```

Comment:
suggestion: Duplicating these here doesn't feel ideal; `./hack/latest-kind-images.sh` writes this data to a file which can be included from a script.

Having these hashes baked into this file feels like going backwards in terms of automation since it currently adds a manual step for updating the base images.

Reply:
Fixed in 4a5ff364258cfbc933a62b1e5e62d9cced0023b1.